### DISCUSSION TITLE
Adding VAT number field and checkbox if user want recipt or invoice

### DISCUSSION THREAD

**Discussion by alzajac** [woocommerce/woocommerce#45079] [WooCommerce, Blocks, and Site Editing]
Hello, there is a chance to checkout new field for VAT number and a checkbox billing document type - if user wants a recipt or invoice. If user want invoice then field for VAT number is showing, but if user will check recipt then field for VAT number is hidden. This should be also related to "company" field - this field should also be shown only if customer will want an invoice. You could add a validation for VAT EU number but it should be only an option - if somone is selling only in his country it is cousing more problems for customer because he has to add country prefix before VAT number - now everyone knows about it.

**Comment by senadir** [comment]
Hey, we're aware of this limitation and are working on fixing it in the near future. You can see previous reports of this issue 
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3096
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3499
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4116

I'm going to close this issue because we're already tracking it, you can leave any additional comments on this issue.

**Reply by mojzesz777** [reply]
@senadir I can see that you are working... over 3 years and nothing.

**Comment by nielslange** [comment]
Hello @alzajac 👋

I've shared an example on how to add a VAT field to the Checkout block [here](https://github.com/woocommerce/woocommerce/discussions/51177#discussioncomment-11276189). Currently, implementing conditional field display isn't straightforward. If this functionality is crucial for your setup, please consider following the steps outlined in our [Tutorial: Extending the WooCommerce Checkout Block](https://developer.woocommerce.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/). I hope this guide helps you achieve the desired outcome.
